src/projectile.h

```diff
@@ -62,12 +64,16 @@ int getExpGain(int player);
 /// Calculate the initial velocities of an indirect projectile. Returns the flight time.
 int32_t projCalcIndirectVelocities(const int32_t dx, const int32_t dz, int32_t v, int32_t *vx, int32_t *vz, int min_angle);
 
-/** Send a single projectile against the given target. */
-bool proj_SendProjectile(WEAPON *psWeap, SIMPLE_OBJECT *psAttacker, int player, Vector3i target, BASE_OBJECT *psTarget, bool bVisible, int weapon_slot);
+/** Send a single projectile against the given target.
+  * Returns a non-null pointer to the newly-created projectile in the case of penetrating projectiles.
+  * The returned projectile needs to be manually added `psProjectileList` global list. */
+PROJECTILE* proj_SendProjectile(WEAPON *psWeap, SIMPLE_OBJECT *psAttacker, int player, Vector3i target, BASE_OBJECT *psTarget, bool bVisible, int weapon_slot);
 
 /** Send a single projectile against the given target
- * with a minimum shot angle. */
-bool proj_SendProjectileAngled(WEAPON *psWeap, SIMPLE_OBJECT *psAttacker, int player, Vector3i target, BASE_OBJECT *psTarget, bool bVisible, int weapon_slot, int min_angle, unsigned fireTime);
+ * with a minimum shot angle.
+ * Returns a non-null pointer to the newly-created projectile in the case of penetrating projectiles.
+ * The returned projectile needs to be manually added `psProjectileList` global list. */
+PROJECTILE* proj_SendProjectileAngled(WEAPON *psWeap, SIMPLE_OBJECT *psAttacker, int player, Vector3i target, BASE_OBJECT *psTarget, bool bVisible, int weapon_slot, int min_angle, unsigned fireTime);
```

Comment:
@ManManson It seems like all the uses of `proj_SendProjectile` and `proj_SendProjectileAngled` _outside_ of `projectile.cpp` always call `proj_AddActiveProjectile` afterwards.

So perhaps we should return the public functions to always doing so (and returning bool), and add new private functions in `projectile.cpp` (example: `proj_SendProjectileInternal`?) which use this new signature, and are only used inside of `projectile.cpp`.

(To keep the extra logic of adding the projectile to the active list contained in a single place, and eliminate the possibility of mistaken omissions elsewhere.)

Reply:
Totally makes sense, thanks for pointing this out. Will fix.

Reply:
Done.